### ISSUE TITLE
OpenSSL 3.0 beta 1: How to confirm that an EVP_PKEY operation is being provided by the FIPS module

### ISSUE THREAD

**Issue by dskern-github** [openssl/openssl#16058]
Section 7.8 of the OpenSSL 3.0 wiki page https://wiki.openssl.org/index.php/OpenSSL_3.0 and https://www.openssl.org/docs/manmaster/man7/fips_module.html each show a similar series of calls that can be used to determine if a message digest operation is being provided by the FIPS module. Equivalent calls also exist for EVP_CIPHER_CTXs. 

However, there does not appear to be an `EVP_PKEY_get0_provider` - type function in OpenSSL 3.0 beta 1 that could be used to follow this chain for an asymmetric operation.  Was this an oversight, or is there a different approach that should be used for asymmetric operations?

**Comment by paulidale** [comment]
Question for OTC: should there be an easy way to determine a provider from an EVP_PKEY and EVP_PKEY_CTX?

The difficulty is where this information comes from.

For EVP_PKEY, the EVP_KEYMGMT is one possibility but the operation cache links to other providers and all could be considered related.

For EVP_PKEY_CTX, the EVP_KEYMGMT again exists but so do EVP_ASYM_CIPHER, EVP_KEYEXCH, EVP_KEM and EVP_SIGNATURE.

**Comment by levitte** [comment]
For an EVP_PKEY, the provider would be the one from its EVP_KEYMGMT
For an EVP_PKEY_CTX, the provider would be the one from the operation implementation (EVP_ASYM_CIPHER, EVP_SIGNATURE, ...).

**Comment by t8m** [comment]
It would be nice to have EVP_PKEY_CTX_get0_provider() to query the provider of an operation. Of course it would return meaningful (non-NULL) value only after the operation was initialized.

**Comment by paulidale** [comment]
OTC: EVP_PKEY_get0_provider() and EVP_KEY_CTX_get0_provider() calls can be added for 3.0.
Also note that these calls are have not been mandated to be part of 3.0 by the OTC vote.

**Comment by levitte** [comment]
PR #16063
